lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.ts

```diff
@@ -281,11 +352,126 @@ export class ContentMetadataComponent implements OnChanges, OnInit, OnDestroy {
         this.categoryControlVisible = false;
     }
 
-    cancelChanges() {
+    cancelChanges(buttonType: ButtonType, event: MouseEvent, group?: CardViewGroup) {
+        event.stopPropagation();
+        this.toggleEditMode(buttonType, group);
         this.revertChanges();
         this.loadProperties(this.node);
     }
 
+    // Returns the editing state of the panel
+    isEditingPanel(): boolean {
+        return (
+            (this.editable && this.hasMetadataChanged) ||
+            (this.editableTags && this.hasMetadataChanged) ||
+            (this.editableCategories && this.hasMetadataChanged) ||
+            (this.editableGroup?.editable && this.hasMetadataChanged)
+        );
+    }
+
+    showSnackbarError(message: string): void {
```

Comment:
Are you using this function anywhere else than in `toggleEdit`? If not you can just call it inside the function there is no need for separate method

Reply:
Only inside the toggleEdit method and we have spyon this method in the test case. Is it okay to have a separate method?

Reply:
You can spy on `showError` from `notificationService` to get the same result and since you're not using this method elsewhere I would remove it

Reply:
Removed the method.